### ISSUE TITLE
minimumDisplayTime not being honored for dismiss while presenting

### ISSUE THREAD

**Issue by mrubin** [AssistoLab/KVNProgress#87]
If you call `dismissWithCompletion()` while the view is still animating on from the `show()`, `minimumDisplayTime` is not honored. We're hitting the return in the below snippet

```
+ (void)dismissWithCompletion:(KVNCompletionBlock)completion
{
	if ([self sharedView].state == KVNProgressStateHidden) {
		return;
	} else if ([self sharedView].state == KVNProgressStateAppearing) {
		[self sharedView].state = KVNProgressStateDismissing;
		[[self sharedView].animateAppearanceOperation cancel];
		[self endDismissWithCompletion:completion];
		
		return;
	}
```

and we never get into `dismissAnimatedWithCompletion()`, which is where `minimumDisplayTime()` is checked.

**Comment by kevin-hirsch** [comment]
Hello @mrubin,

I see your point here. The philosophy behind the `minimumDisplayTime` actually is that if the loader is shown to the user, we use the `minimumDisplayTime` property to make it stay long enough on screen and not blink. But if the loader has not appeared yet, there's no need to show it for a certain amount of time in the first place.

Now I see your point, and reading what I've put in the *README.md* actually go towards what you're saying:
> To avoid the user to see a blinking HUD or even **don't see it at all** if you are dismissing it too quickly, the HUD will stay display for a minimum (short) period of time.


I could investigate to check if the change would be easy to do. But I think I remember I struggled with the state of the loader because `show` and `dismiss` methods are asynchronous. 
So in the case you're describing, the code should remember that a dismiss should be actually called after the loader is actually really shown on screen. I will check it out when I have more time. Even though I don't think this is a priority (or is it on your side?).

Thanks for noticing 👍
I hope you enjoy using `KVNProgress`!

**Comment by mrubin** [comment]
Thanks for the quick response!

`show` & `dismiss` are indeed a big pain, especially when you're dealing with edge-cases like the app resigning/becoming active and/or going to/from background. In my app, a server response can cause an alert to pop-up at any time. Also, if the app becomes active again (e.g. accidentally toggle Control Center or Split Screen for a moment, or return from background) and we have to essentially pause the app and put up a 'Logging In' overlay until we can re-authenticate with our server. This can lead to all sorts of scenarios where you have modals on top of modals (loading overlays on top of alerts, or alerts on top of loading overlays, or a 'logging in' loading overlay on top of another loading overlay, etc.). In some cases - e.g. if you accidentally present on top of an alert that's in the process of dismissing - your VC might not present at all... which, in case you were waiting for its completion to run (if it's an alert, for example.. and the user has to choose an option), causes you to get stuck.

I was interested in how `KVNProgress` deals with using a separate `UIWindow` for presenting these modals. It seems to work really well in a lot of cases... one use-case that doesn't seem to be supported (since the model is essentially that of a singleton) is presenting multiple modals on top of each other. Though I understand that this is usually not desired... :) But it's something my app has to accommodate.

**Comment by kevin-hirsch** [comment]
Indeed, the use case of several modals is not handled by `KVNProgress` but the way I would do it is like with `UIView`s. I think your windows with the same `windowLevel` will stack on top of each other in the order you add them to the hierarchy (via `makeKeyAndVisible`) like `UIView`s do when the have the same `zIndex` property.
